cups-toshiba-estudio: init at 7.51

jpotier opened the pull request:
###### Motivation for this change
There was no driver for the toshiba-estudio series of printers. This is a generic driver that supports many
printers (see comment in the `default.nix`).

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested function of the printer driver (I've used it at work for a few weeks now)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Commit:
cups-toshiba-estudio: init at 7.51

zimbatm commented on pkgs/misc/cups/drivers/estudio/default.nix:
```diff
@@ -0,0 +1,78 @@
+# TOSHIBA e-STUDIO6550C
+# TOSHIBA e-STUDIO6560C
+# TOSHIBA e-STUDIO6570C
+# TOSHIBA e-STUDIO7506AC
```
It might make sense to put that in the meta.longDescription to make it searchable as a user.

jpotier replied:
Ok, makes sense.

zimbatm commented on pkgs/misc/cups/drivers/estudio/default.nix:
```diff
@@ -0,0 +1,78 @@
+             "installPhase" ];
+
+  preInstall = ''
+    ls
```
debug leftover?

jpotier replied:
Ooops, didn't see this one

zimbatm commented on pkgs/misc/cups/drivers/estudio/default.nix:
```diff
@@ -0,0 +1,78 @@
+  preInstall = ''
+    ls
+    patchShebangs lib/
+    head lib/cups/filter/est6550_Authentication
```
debug leftover?

jpotier replied:
Yeah, of it goes

zimbatm commented on pkgs/misc/cups/drivers/estudio/default.nix:
```diff
@@ -0,0 +1,78 @@
+    head lib/cups/filter/est6550_Authentication
+
+    gunzip                share/cups/model/Toshiba/TOSHIBA_ColorMFP_CUPS.gz
+    sed -i "s+/usr+$out+" share/cups/model/Toshiba/TOSHIBA_ColorMFP_CUPS
```
is there only one /usr to patch?

jpotier replied:
Yes, any thought on how to do it better?

zimbatm commented on pkgs/misc/cups/drivers/estudio/default.nix:
```diff
@@ -0,0 +1,78 @@
+             "preInstall"
+             "installPhase" ];
+
+  preInstall = ''
```
can you call this the configurePhase or patchPhase? pre-phases are supposed to be called from the phase itself

jpotier replied:
Let's go for patchPhase

zimbatm reviewed (commented):
I don't have a printer available to test this out. How did you test the package?

zimbatm commented on pkgs/misc/cups/drivers/estudio/default.nix:
```diff
@@ -0,0 +1,78 @@
+    description = "Printer only driver for the Toshiba e-STUDIO class of printers";
+    homepage = https://www.toshiba-business.com.au/support/drivers;
+    license = stdenv.lib.licenses.unfree;
+    maintainers = with stdenv.lib.maintainers; [ jpotier ];
```
adding `with stdenv.lib;` on the whole meta block removes a bit of duplication, not required though

jpotier replied:
Looks cleaner, will do

jpotier commented:
About testing: I've first built it as an independant package and I manually selected the PPD when installing the printer with CUPS. Then, I forked and cloned nixpkgs locally and added the package (this commit). I added the printer driver in driver in `printing.drivers` list in my `configuration.nix`. I reinstalled the printer, checked it was now available in the cups menu and tested printing again.

Commit:
cups-toshiba-estudio: init at 7.51 (including reviews)
